File: Desktop/Viewer/src/main/kotlin/com/example/demo/app/PictureModule/Contents/Audio.kt
package com.goldenratio.onepic.PictureModule.Contents


class Audio (audioByteArray : ByteArray? = null, _Content_attribute: ContentAttribute){
    var _audioByteArray : ByteArray? = null
    var attribute = _Content_attribute
    var size : Int = audioByteArray?.size ?: 0
    init {
        if (audioByteArray != null) {
            _audioByteArray = audioByteArray
            size = audioByteArray!!.size
        }
        attribute = _Content_attribute
    }

    fun waitForByteArrayInitialized() {
        while (!isByteArrayInitialized()) {
            Thread.sleep(100)
        }
    }

    fun isByteArrayInitialized(): Boolean {
        return _audioByteArray != null
    }
//    fun getInfoLength() : Int{
//        // offset(4) + attribute(4) + size(4)
//        return 16
//    }
}

File: Chorme_extension/contentScript/All-in-JPEG/content/Audio.js
export default class Audio{
    constructor(audioByteArray, _Content_attribute) {
        this._audioByteArray = null;
        this.attribute = _Content_attribute;
        this.size = audioByteArray ? audioByteArray.length : 0;

        if (audioByteArray) {
            this._audioByteArray = new Uint8Array(audioByteArray);
            this.size = audioByteArray.length;
        }
    }

    async waitForByteArrayInitialized() {
        while (!this.isByteArrayInitialized()) {
            await new Promise(resolve => setTimeout(resolve, 100));
        }
    }

    isByteArrayInitialized() {
        return this._audioByteArray !== null;
    }
}
